File: app/src/main/java/com/kkkkkn/readbooks/view/activities/BaseManager.java
package com.kkkkkn.readbooks.view.activities;

import android.util.Log;

import androidx.annotation.NonNull;
import androidx.lifecycle.Lifecycle;
import androidx.lifecycle.LifecycleEventObserver;
import androidx.lifecycle.LifecycleOwner;

public class BaseManager implements LifecycleEventObserver {
    private static final String TAG="BaseManager";

    @Override
    public void onStateChanged(@NonNull LifecycleOwner source, @NonNull Lifecycle.Event event) {
        switch (event){
            case ON_CREATE:
                Log.i(TAG, "onStateChanged: ON_CREATE");
                break;
            case ON_DESTROY:
                Log.i(TAG, "onStateChanged: ON_DESTROY");
                break;
        }
    }


}

File: app/src/main/java/com/kkkkkn/readbooks/view/activities/BaseManager.kt
package com.kkkkkn.readbooks.view.activities

import android.util.Log
import androidx.lifecycle.Lifecycle
import androidx.lifecycle.LifecycleEventObserver
import androidx.lifecycle.LifecycleOwner

class BaseManager : LifecycleEventObserver {
    override fun onStateChanged(source: LifecycleOwner, event: Lifecycle.Event) {
        when (event) {
            Lifecycle.Event.ON_CREATE -> Log.i(TAG, "onStateChanged: ON_CREATE")
            Lifecycle.Event.ON_DESTROY -> Log.i(TAG, "onStateChanged: ON_DESTROY")
            else -> {}
        }
    }

    companion object {
        private const val TAG = "BaseManager"
    }
}